masim.c

```diff
@@ -312,10 +322,18 @@ void exec_config(struct access_config *config)
 {
 	struct mregion *region;
 	size_t i;
+	void *addr;
 
 	for (i = 0; i < config->nr_regions; i++) {
 		region = &config->regions[i];
 		region->region = (char *)malloc(region->sz);
+		if (hugepage) {
+			addr = mmap(region->region, region->sz, PROTECTION, FLAGS, -1, 0);
+			if (addr == MAP_FAILED) {
+				perror("mmap");
+				exit(1);
+			}
+		}
```

Comment:
Where `addr` is being used?  I guess `malloc()` shouldn't be called, and `addr` should be assigned to `region->region` if `hugepage` is not NULL?  Also, `addr` should be unmapped at last.

I thought I made this very same comment to the previous revision, but seems I messed up something.

Reply:
yes you are right, I'll correct this